[TEST] add unit tests for ticket-transfer doctype

harshtandiya opened the pull request:
## What type of PR is this? (check all applicable)
- [x] Test
## Description
This PR adds a couple of unit tests for the Ticket Transfer Doctype.
- Testing Ticket Transfer Workflow ->
   - Check that ticket is created only for a valid event.
   - Check that once the transfer is done, there is no ticket remaining with the details of the previous owner
   - Verify that the ticket has the details of the new owner after transfer 
- Testing Transfer Status to be "Pending Approval" at the time of creation, and that it throws a ValidationError if it is anything else.

```
----------------------------------------------------------------------
Ran 2 tests in 1.889s

```

harshtandiya committed:
test(ticket-transfer): test ticket transfer workflow

harshtandiya committed:
test(ticket-transfer): add test to check transfer status on creation

NagariaHussain commented on fossunited/ticketing/doctype/foss_event_ticket_transfer/test_foss_event_ticket_transfer.py:
```diff
@@ -1,8 +1,119 @@
+        self.assertEqual(transfer.status, "Pending Approval")
+
+        # Change status of transfer to completed
+        transfer.status = "Completed"
+        transfer.save()
+        self.assertEqual(transfer.status, "Completed")
```
You don't have to test the framework 😅

rahulporuri replied:
this might be a left over assert - at first, we were using a method on the doctype to set the status and perform the validation internally. we ran into some issues there so this assert can go now.

NagariaHussain commented on fossunited/ticketing/doctype/foss_event_ticket_transfer/test_foss_event_ticket_transfer.py:
```diff
@@ -1,8 +1,119 @@
+        )
+        self.assertTrue(new_ticket_exists)
+
+        transfer.delete()
+        ticket.delete()
+        event.delete()
```
Rollback automatically happens...

rahulporuri replied:
after each test or after the entire testsuite?

we wanted to ensure that we don't have potential test interactions between unit tests, which is why we went ahead and deleted them manually

rahulporuri commented on fossunited/ticketing/doctype/foss_event_ticket_transfer/test_foss_event_ticket_transfer.py:
```diff
@@ -1,8 +1,119 @@
+        ticket.delete()
+        event.delete()
+
+    def test_check_status_pending_on_create(self):
```
nitpicky comment - we might need a better name because it wasn't obvious what we were testing until i read the test code. but i cant come up with a better name myself so 🤷

harshtandiya replied:
`test_status_pending_on_create`?

harshtandiya committed:
chore(ticket-transfer): remove redundant test code

harshtandiya committed:
chore(ticket-transfer): rename test function

rename `test_check_status_pending_on_create` -> `test_status_pending_on_create` as the prior was

rahulporuri reviewed (approved):
LGTM. I think it's worth waiting for @NagariaHussain s review for a day or two. We're not in a hurry to merge this PR.

rahulporuri commented:
@NagariaHussain we're going ahead and merging this PR for now. Please do take a look at it post-merge, if you have the time. we are still trying our best to understand the best practices around unittesting so your comments in this regard are incredibly valuable
